### PR TITLE
Prevent the reconcile loop from updating the services on every run

### PR DESCRIPTION
**Pull request by bdunne**

### CODE REVIEW

File: manageiq-operator/pkg/helpers/miq-components/memcached.go [hunk 1]
```diff
@@ -101,19 +101,18 @@ func NewMemcachedService(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme) (*cor
...
 			return err
 		}
 		addAppLabel(cr.Spec.AppName, &service.ObjectMeta)
-		service.Spec.Ports = []corev1.ServicePort{
-			corev1.ServicePort{
-				Name: "memcached",
-				Port: 11211,
-			},
-		}
+		service.Spec.Ports[0].Name = "memcached"
```
Comment (carbonin) [inline manageiq-operator/pkg/helpers/miq-components/memcached.go:114]: I'm worried that his `Ports` array will come back as `nil` if there isn't anything set. Can you check for nil here?

Comment (carbonin) [reply manageiq-operator/pkg/helpers/miq-components/memcached.go:114]: Also it seems like it wouldn't be worth setting it in the service spec above, right?

Comment (bdunne) [reply manageiq-operator/pkg/helpers/miq-components/memcached.go:114]: Nope, turns out I don't need that

Comment (bdunne) [reply manageiq-operator/pkg/helpers/miq-components/memcached.go:114]: Fixed the nil issue and tested that it gets recreated is the service is deleted

**Comment by carbonin** [comment]
Looks good, but what was the problem @bdunne? What were we touching that we shouldn't have been?

**Comment by bdunne** [comment]
Previously:
`{"level":"info","ts":1594070650.2048993,"logger":"controller_manageiq","msg":"Service has been reconciled","component":"memcached","result":"updated"}`
Now:
`{"level":"info","ts":1594146078.004267,"logger":"controller_manageiq","msg":"Service has been reconciled","component":"memcached","result":"unchanged"}`

Something on the ServicePort was being overwritten causing the service to be updated on every loop.

**Comment by carbonin** [comment]
Oh, sure. I was just curious what exactly the server what changing between reconcile calls.

**Comment by bdunne** [comment]
I'm not sure what adds the `TargetPort`, maybe it's removing that and the server adds it back in?

**Pull request merged**